astropy/wcs/tests/test_wcs.py

```diff
@@ -914,3 +914,26 @@ def test_no_truncate_using_compare():
     w.wcs.set()
     w2 = wcs.WCS(w.to_header())
     w.wcs.compare(w2.wcs)
+
+
+def inconsistent_sip():
+    """
+    Test for #4814
+    """
+    hdr = get_pkg_data_contents("data/sip-broken.hdr")
+    w = wcs.WCS(hdr)
+    newhdr = w.to_header(relax=None)
+    # CTYPE should not include "-SIP" if relax is None
+    assert(all([ctyp[-4 :] != '-SIP' for ctyp in self.wcs.ctype]))
+    newhdr = w.to_header(relax=False)
+    # CTYPE should not include "-SIP" if relax is False
+    assert(all([ctyp[-4 :] != '-SIP' for ctyp in self.wcs.ctype]))
+    newhdr = w.to_header(key="C")
+    # Test writing header with a different key
+    assert(all([ctyp[-4 :] != '-SIP' for ctyp in self.wcs.ctype]))
+    newhdr = w.to_header(key=" ")
+    # Test writing a primary WCS to header
+    assert(all([ctyp[-4 :] != '-SIP' for ctyp in self.wcs.ctype]))
+    # Test that "-SIP" is kept into CTYPE if relax=True
+    newhdr = w.to_header(relax=True)
+    assert(all([ctyp[-4 :] == '-SIP' for ctyp in self.wcs.ctype]))
```

Comment:
@nden - in the above tests, can you also add a test of whether one of the SIP coefficients is present or not in the resulting header?